tests/epyccel/recognised_functions/test_numpy_operators.py

```diff
@@ -0,0 +1,204 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring
+import os
+from typing import TypeVar
+from numpy.random import randint
+import numpy as np
+import pytest
+
+from pyccel import epyccel
+
+int_types = (bool, np.int8, np.int64)
+IT = TypeVar('IT', *int_types)
+IT2 = TypeVar('IT2', *int_types)
+
+def test_numpy_bit_and(language):
+    def f(a : 'IT[:,:,:]', b : 'IT2[:,:,:]'):
+        return a & b
+    def g(a : 'IT[:,:,:]', b : 'IT2'):
+        return a & b
+    def h(a : 'IT', b : 'IT2[:,:,:]'):
+        return a & b
+
+    epyc_f = epyccel(f, language=language)
+    epyc_g = epyccel(g, language=language)
+    epyc_h = epyccel(h, language=language)
+
+    for t_x in int_types:
+        for t_y in int_types:
+            x = randint(2 if t_x is bool else 127, size=(2,3,4), dtype = t_x)
+            y = randint(2 if t_y is bool else 127, size=(2,3,4), dtype = t_y)
```

Comment:
Should we check a case that requires broadcasting? Here we could set to `1` any of the dimensions of the array `y`.

Reply:
Indeed, we could drop the first dimension.

Not related to this PR: why is the case of 1 not working, given that we can already drop one dimension? In any case we must check the dimensions of the arrays at runtime